lib/skylight/normalizers/coach/handler_finish.rb

```diff
@@ -0,0 +1,52 @@
+module Skylight
+  module Normalizers
+    module Coach
+      class HandlerFinish < Normalizer
+        register "coach.handler.finish"
+
+        CAT = "app.coach.handler".freeze
+
+        # See information on the events Coach emits here:
+        # https://github.com/gocardless/coach#instrumentation
+
+        # Run when the handler first starts, we need to set the trace endpoint to be the
+        # handler name.
+        #
+        # We can expect the payload to have the :middleware key.
+        def normalize(trace, name, payload)
+          trace.endpoint = payload[:middleware]
+          [ CAT, payload[:middleware], nil ]
+        end
```

Comment:
So if we were to setup an endpoint like so:

```ruby
class Authenticate < Coach::Middleware
  def call
    next_middleware.call
  end
end

class Whoami < Coach::Middleware
  uses Authenticate 
  
  def call
    [200, {}, []]
  end
end

handler = Coach::Handler.new(Whoami)
handler.call
```

Then `coach.middleware.finish` events as each middleware runs. In this case, we'd see (and therefore set the trace endpoint to be) `Authenticate` and then `Whoami`. When you open Skylight, you want to be looking at an endpoint named after your handler (you'd expect `Whoami` here) and for there to be a trace that you can view that shows you the full runtime of the `Whoami` handler.

Without setting the `trace.endpoint` in response to the `coach.handler.finish` event, I don't think we'd end up with a trace in the UI that can easily give us the time for the entire handler, separate from the previous Rack layer. I reckon that's probably useful to have, so wouldn't mind keeping it!

I've included a screenshot below with a red arrow to show where this appears- hopefully you could see why this would be useful to identify the handler duration as a whole.

![screen shot 2017-08-07 at 21 03 40](https://user-images.githubusercontent.com/3518874/29069680-c90a3f40-7c33-11e7-8f83-4f167461730f.jpg)